### ISSUE TITLE
In preparation for the new EasyHook: IPC implementation and possibility to make an unified hooking service

### ISSUE THREAD

**Issue by netcorefan1** [EasyHook/EasyHook#362]
Hello to all,
recently @spazzarama started working on the new [NetStandard](https://github.com/EasyHook/EasyHook/tree/feature/netstandard) branch. The new version will be compatible with .Net core and .Net 5. The IPC functionality will be removed leaving to the users the possibility to choose from the variety of other awesome IPC solutions out there. More info on the current status of the project [here](https://github.com/EasyHook/EasyHook/issues/278).

I am actually testing several IPC solutions and interested people can see my recent opened issues to get more info. Actually I didn't come to a decision, but nanomsg-next-gen ([nng.NETCore](https://github.com/jeikabu/nng.NETCore)) is the one where I actually saw very high performance (only problem is that is very difficult to use or at least requires a long learning curve). It seems to be the more flexible in terms of customizations and the more appreciated from the users.
If the community all agree to use one specific IPC solution it will be of benefits for all because we can help each other.

With the new version of EasyHook it came to me the idea to make an unified hooking service where applications can connect and enable only the specific hooks they need. Until now I created different projects for different uses cases and this is something that I don't like at all. It would be nice having EasyHook as some sort of central station, may be under a Windows Service where we connect from the app and execute the custom hooks (methods, callbacks etc). I am not sure if this is feasible at all and in what way could be implemented.
Hope to see the new EasyHook soon in order to start playing!
Any suggestion are welcome!

**Comment by spazzarama** [comment]
Yes something like this could be done, and in fact for kernel mode hooking this is more or less how you have to implement it.

I'm not sure I'm seeing the value you are for a central service to provide hooking services. Hooking generally is for very specific use cases.

Perhaps instead we can add more helpers within the existing library that simplify things, or allow hooks to be described in some form that allow other libraries to consume easily.
